src/clj/puppetlabs/services/jruby/jruby_agents.clj

```diff
@@ -62,13 +61,14 @@
   "Flush a single JRubyInstance.  Create a new replacement instance
   and insert it into the specified pool."
   [pool-context :- jruby-schemas/PoolContext
-   instance :- JRubyInstance
+   instance :- jruby-schemas/JRubyInstanceSchema
    new-pool :- jruby-schemas/pool-queue-type
    new-id :- schema/Int
    config :- jruby-schemas/JRubyConfig]
-  (jruby-internal/cleanup-pool-instance! instance)
-  (jruby-internal/create-pool-instance! new-pool new-id config
-                                        (partial send-flush-instance! pool-context)))
+  (let [shutdown-fn (get-in pool-context [:config :lifecycle :shutdown])]
```

Comment:
Do you think we could rename this lifecycle function to `:cleanup`?  Or `:terminate`?  The term `shutdown` is kind of overloaded in TK so I was a little thrown off by these references to "shutdown" at first.